content/concepts/multiplex/overview.md

```diff
@@ -9,24 +9,36 @@ aliases:
 
 ## Overview
 
-Stream Multiplexing (_stream muxing_) is a way of sending multiple streams of data over one
-communication link. It combines multiple signals into one unified signal so it can be transported
-'over the wires', then it is demulitiplexed (_demuxed_) so it can be output and used by separate
-applications. This is done to share a single TCP connection using unique port numbers to distinguish
-streams, between the multiple proceeses (such as kademlia and gossipsub) used by applications (such as IPFS)
-to make connection and transmission more efficient.
-
-With muxing, libp2p applications may have many separate streams of communication between peers, as well as
-have multiple concurrent streams open at the same time with a peer. Stream multiplexing allows us to initialize
-and use the same [transport](../../transports/overview) connection across the lifetime of our interaction with a
-peer. With muxing, we also only need to deal with [NAT traversal](../../nat/overview) once to be able to open as
-many streams as we need, since they will all share the same underlying transport connection. Applications can enable
-support for multiple multiplexers, which will allow you to fall back to a widely-supported multiplexer if a preferred
-choice is not supported by a remote peer.
+Stream multiplexing (or stream muxing) is a method to send multiple streams of
+data over a single communication link. It allows peers to use the same connection
+across the lifetime of their interaction with another peer.
+
+In particular, stream muxing is done to share a single connection between multiple
+protocols used by applications to make connection and transmission more efficient.
+This is especially useful when doing NAT traversal or hole punching,
+as it allows for efficient use of limited resources. This is also used outside of
+libp2p, for example, HTTP/2 introduced streams into HTTP, allowing for many HTTP
+requests in parallel on the same connection.
```

Comment:
I agree. This document should also be prefaced with a paragraph on why stream multiplexing is necessary. It hints at it but doesn't explicitly mention why it's critical.

I would add:

libp2p is built on top of a stream abstraction (link to the protocols page) and uses a bi-directional message stream to send data between peers.
However, simply having a single message stream over a connection between two peers is not good enough.
This limits the scalability of applications and in fact creates a bottleneck.
This is because each peer on the other side of the connection could be running multiple applications, each sending and waiting for messages over the single stream.

With only a single stream to use, application A would have to wait for application B's use of the connection to end before another A could start sending and receiving messages over the lone stream.

Therefore, to eliminate this bottleneck and allow more applications to talk to each other over the single connection, libp2p employs stream multiplexing.
This allows a libp2p node to create many virtual connections inside a single connection and allows a node to send multiple different streams of messages over those different virtual connections.

Reply:
It was previously suggested to avoid the mention of connections. Agreed though, this is important.